sstable/src/delta.rs

```diff
@@ -45,13 +46,33 @@ where
             return Ok(None);
         }
         let start_offset = self.write.written_bytes() as usize;
+
         let buffer: &mut Vec<u8> = &mut self.stateless_buffer;
         self.value_writer.serialize_block(buffer);
         self.value_writer.clear();
+
         let block_len = buffer.len() + self.block.len();
-        self.write.write_all(&(block_len as u32).to_le_bytes())?;
-        self.write.write_all(&buffer[..])?;
-        self.write.write_all(&self.block[..])?;
+
+        if block_len > 2048 {
+            buffer.extend_from_slice(&self.block);
+            self.block.clear();
+            zstd::stream::copy_encode(&**buffer, &mut self.block, -1)?;
```

Comment:
Maybe this would benefit from using `Compressor::compress_to_buffer` as well to avoid the `io::Read/Write` based interfaces? Resizing an amortized output buffer should be possible using [`zstd::zstd_safe::compress_bound`](https://docs.rs/zstd-safe/6.0.3+zstd.1.5.2/zstd_safe/fn.compress_bound.html) as `Compressor::compress` does internally.

Reply:
When looking at flamegraphs, this part seemed fine for some reason. I rewrote it with `compress_bound` and the bulk api. I can't tell if it's any faster, results are within margin of error. However it seems to yield a more compact representation so I'll do the change anyway

Reply:
> However it seems to yield a more compact representation so I'll do the change anyway

This might be the difference between stream and bulk API where the bulk API knows all of its input including the size upfront.

Reply:
yeah, and I guess this different encoding is probably helpful to `Decompressor::upper_bound` too